src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -385,5 +475,272 @@ private void FillValues(int srcLength, ref VBuffer<bool> dst, List<int> indices,
                 dst = new VBuffer<bool>(srcLength, dstValues, dstIndices);
             }
         }
+
+        protected override IRowMapper MakeRowMapper(ISchema schema)
+            => new Mapper(this, schema);
+
+        private sealed class Mapper : MapperBase
+        {
+            private sealed class ColInfo
+            {
+                public readonly string Name;
+                public readonly string Source;
+                public readonly ColumnType TypeSrc;
+
+                public ColInfo(string name, string source, ColumnType type)
+                {
+                    Name = name;
+                    Source = source;
+                    TypeSrc = type;
+                }
+            }
+
+            private readonly NAIndicatorTransform _parent;
+            private readonly ColInfo[] _infos;
+            private readonly ColumnType[] _types;
+            // The isNA delegates, parallel to Infos.
+            private readonly Delegate[] _isNAs;
+
+            public Mapper(NAIndicatorTransform parent, ISchema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _infos = CreateInfos(inputSchema);
```

Comment:
I don't see a reason to have `CreateColInfos`. Just put this logic here, and merge with the loop below